src/slg/film/imagepipeline/plugins/intel_oidn.cpp

```diff
@@ -30,73 +31,84 @@ using namespace slg;
 
 BOOST_CLASS_EXPORT_IMPLEMENT(slg::IntelOIDN)
 
-IntelOIDN::IntelOIDN(const u_int n, const u_int o, const u_int t, const bool b) {
-	nTiles = n;
-	pixelOverlap = o;
-	pixelThreshold = t;
+IntelOIDN::IntelOIDN(const u_int n, const bool b) {
+	nPixels = n;
 	benchMode = b;
 }
 
 IntelOIDN::IntelOIDN() {
-	nTiles = 4;
-	pixelOverlap = 48;
-	pixelThreshold = 8294400;
+	nPixels = 1000;
 	benchMode = false;
 }
 
 ImagePipelinePlugin *IntelOIDN::Copy() const {
-    return new IntelOIDN(nTiles, pixelOverlap, pixelThreshold, benchMode);
+    return new IntelOIDN(nPixels, benchMode);
 }
 
 void IntelOIDN::Apply(Film &film, const u_int index) {
 	const u_int width = film.GetWidth();
 	const u_int height = film.GetHeight();
 	const u_int pixelCount = width * height;
+
+	bool splitCondition = false;
+	if ((width > nPixels) || (height > nPixels)) {
+		iTileCount = ceil(1.0 * height / nPixels);
+		jTileCount = ceil(1.0 * width / nPixels);
```

Comment:
I don't really understand why you multiply with 1.0 here, is this some secret trick?